cmd/dep/main.go

```diff
@@ -153,6 +166,26 @@ func (c *Config) Run() int {
 		return successExitCode
 	}
 
+	// 'dep help constraints' generates documentation on the exact syntax for version constraints
+	if printCommandHelp && cmdName == "constraints" {
+		fmt.Println("Version constraint syntax:")
+		fmt.Println()
+		firstSentence := `"dep ensure" uses an external semver (https://github.com/Masterminds/semver) `
+		firstSentence += "library to interpret the version constraints you specify in the manifest."
+		fmt.Println(firstSentence)
+		fmt.Println()
+		fmt.Println("The comparison operators are:")
+		fmt.Println()
+
+		for _, constraint := range versionConstraints {
+			fmt.Println(fmt.Sprintf("  %-5s  %s", constraint[0], constraint[1]))
+		}
+
+		fmt.Println()
+		fmt.Println(`Use "dep help [command]" for more information about a command.`)
+		return successExitCode
+	}
```

Comment:
I think it would be better to have this whole block as a function and pass a writer to the function, similar to [usage](https://github.com/golang/dep/blob/master/cmd/dep/main.go#L94-L116), instead of writing to the stdout always.